arches/app/models/mobile_survey.py

```diff
@@ -369,19 +372,29 @@ def collect_resource_instances_for_couch(self):
             search_res_json = search.search_results(request)
             search_res = JSONDeserializer().deserialize(search_res_json.content)
             try:
-                instances = {hit['_source']['resourceinstanceid']: hit['_source'] for hit in search_res['results']['hits']['hits']}
-                # if we didn't get our limit of resource instances using a spatial filter
-                # let's try to get resource instances that don't have spatial data
-                if len(instances.keys()) < int(self.datadownloadconfig['count']):
-                    request.GET['mapFilter'] = '{}'
-                    request.GET['resourcecount'] = int(self.datadownloadconfig['count']) - len(instances.keys())
+                for res_type in resource_types:
```

Comment:
It looks like iteration is taking place only for non-spatial searches. The iteration block also needs to include the spatial search as well.